zerver/actions/realm_emoji.py

```diff
@@ -22,6 +22,7 @@ def notify_realm_emoji(realm: Realm, realm_emoji: Dict[str, EmojiInfo]) -> None:
     send_event_on_commit(realm, event, active_user_ids(realm.id))
 
 
+@transaction.atomic(durable=True)
```

Comment:
I don't love doing network calls to S3 inside transactions. But it does seem likely fine in this case; `RealmEmoji` is a very low contention table.

Reply:
Because of locking?  It's an `INSERT`, so it doesn't take row locks or block anything.  And it's preferable to leaving the race condition where we have a partially-set-up row and no uploaded file for it, which clients could pull.

Reply:
Yeah, mostly a "minimize lock duration" impulse, because bugs where your lock is too long end up wasting a lot of time discovering/debugging. I think this is OK given the nature of the locks in question.